packages/dds/tree/src/test/feature-libraries/forest-summary/incrementalSummaryBuilder.spec.ts

```diff
@@ -0,0 +1,538 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "node:assert";
+
+import { stringToBuffer } from "@fluid-internal/client-utils";
+import type { IExperimentalIncrementalSummaryContext } from "@fluidframework/runtime-definitions/internal";
+import {
+	SummaryTreeBuilder,
+	type ReadAndParseBlob,
+} from "@fluidframework/runtime-utils/internal";
+// import { SummaryType } from "@fluidframework/driver-definitions";
+import type { IChannelStorageService } from "@fluidframework/datastore-definitions/internal";
+import { SummaryType, type ISnapshotTree } from "@fluidframework/driver-definitions/internal";
+import { validateAssertionError } from "@fluidframework/test-runtime-utils/internal";
+
+import {
+	ForestIncrementalSummaryBuilder,
+	ForestSummaryTrackingState,
+	// eslint-disable-next-line import/no-internal-modules
+} from "../../../feature-libraries/forest-summary/incrementalSummaryBuilder.js";
+import type {
+	EncodedFieldBatch,
+	ChunkReferenceId,
+	TreeChunk,
+	// eslint-disable-next-line import/no-internal-modules
+} from "../../../feature-libraries/chunked-forest/index.js";
+import type {
+	FieldKey,
+	ITreeCursorSynchronous,
+	TreeNodeSchemaIdentifier,
+} from "../../../core/index.js";
+
+/**
+ * Creates a mock incremental summary context for testing.
+ */
+function createMockIncrementalSummaryContext(
+	summarySequenceNumber: number,
+	latestSummarySequenceNumber: number,
+	summaryPath: string = "/test/path",
+): IExperimentalIncrementalSummaryContext {
+	return {
+		summarySequenceNumber,
+		latestSummarySequenceNumber,
+		summaryPath,
+	};
+}
+
+/**
+ * Creates a mock channel storage service for testing.
+ */
+function createMockStorageService(
+	snapshotTree?: ISnapshotTree,
+	blobs: Map<string, string> = new Map(),
+): IChannelStorageService {
+	const mockService: Partial<IChannelStorageService> = {
+		getSnapshotTree: snapshotTree ? () => snapshotTree : undefined,
+		contains: async (path: string) => blobs.has(path),
+		readBlob: async (path: string) => {
+			const blobContents = blobs.get(path);
+			if (blobContents === undefined) {
+				throw new Error(`Blob not found: ${path}`);
+			}
+			return stringToBuffer(blobContents, "utf8"); // Mock implementation
+		},
+		list: async () => [],
+	};
+	return mockService as IChannelStorageService;
+}
+
+function getReadAndParse(blobMap: Map<string, string>): ReadAndParseBlob {
+	return async <T>(id: string): Promise<T> => {
+		const blob = blobMap.get(id);
+		if (blob === undefined) {
+			throw new Error(`Blob not found: ${id}`);
+		}
+		return blob as T;
+	};
+}
+
+describe("ForestIncrementalSummaryBuilder", () => {
+	let builder: ForestIncrementalSummaryBuilder;
+	let testCursor: ITreeCursorSynchronous;
```

Comment:
As noted in https://github.com/microsoft/FluidFramework/pull/24949/files#r2261506172, We should not use this pattern.

Reply:
Updated the pattern to add all setup inside a function.